docker/Dockerfile.cpu

```
@@ -1,31 +1,47 @@
-FROM ubuntu:18.04
+FROM ubuntu:18.04 AS compile-image
```

```suggestion
ARG BASE_IMAGE=ubuntu:18.04
FROM ${BASE_IMAGE} AS compile-image
```

Since the `Dockerfile.cpu` and `Dockerfile.gpu` are basically the same it'd be useful to reduce the duplication of code and just use a `BASE_IMAGE` arg here to swap out the `BASE_IMAGE` with the `gpu` image.

See the official `pytorch` Dockerfile for reference: https://github.com/pytorch/pytorch/blob/master/Dockerfile#L10-L13

---

Done. Now there is single Dockerfile to address cpu and gpu images!
Thanks for point this out!

---

@seemethere Can you confirm if this meets your expectations for requested changes and then resolve this conversation?